include/micro/util/queue.h

```diff
@@ -0,0 +1,14 @@
+#ifndef MICRO_QUEUE_H
+#define MICRO_QUEUE_H
+
+#include <stdlib.h>
+
+struct queue;
```

Comment:
This is just a suggestion, to show that the `struct queue` type is a private type usually the character `_` is added at the beginning. In this case should be 
```suggestion
struct _queue;
```